util/src/io/worker.rs

```diff
@@ -123,7 +123,7 @@ impl Worker {
 impl Drop for Worker {
 	fn drop(&mut self) {
 		trace!(target: "shutdown", "[IoWorker] Closing...");
-		let _ = self.wait_mutex.lock();
+		let _ = self.wait_mutex.lock().unwrap();
 		self.deleting.store(true, AtomicOrdering::Release);
 		self.wait.notify_all();
```

Comment:
It seems to be related to mutex unlocking on wait. It is hard to tell cause the logic inside is quite complex. I'll see if I can create an example and file an issue.